File: software/examples/ruby/example_beep.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_piezo_buzzer'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Piezo Buzzer Bricklet

ipcon = IPConnection.new # Create IP connection
pb = BrickletPiezoBuzzer.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Make 2 second beep
pb.beep 2000

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/java/ExampleBeep.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletPiezoBuzzer;

public class ExampleBeep {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Piezo Buzzer Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletPiezoBuzzer pb = new BrickletPiezoBuzzer(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Make 2 second beep
		pb.beep(2000);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
